src/file.rs

```diff
@@ -196,13 +196,27 @@ fn _canonicalize(path: &Path) -> Result<PathBuf> {
     }
 }
 
+fn is_wsl_absolute(path: &str) -> bool {
+    if !cfg!(target_os = "windows") {
+        return false;
+    }
+    let path = path.strip_prefix("/mnt/").unwrap_or(path);
+
+    let maybe_drive = path.split_once('/').map_or(path, |x| x.0);
+
+    maybe_drive.len() == 1 && matches!(maybe_drive.chars().next(), Some('a'..='z'))
+}
+
 // Fix for issue #581. target_dir must be absolute.
 pub fn absolute_path(path: impl AsRef<Path>) -> Result<PathBuf> {
-    Ok(if path.as_ref().is_absolute() {
-        path.as_ref().to_path_buf()
-    } else {
-        env::current_dir()?.join(path)
-    })
+    let as_ref = path.as_ref();
+    Ok(
+        if as_ref.is_absolute() || is_wsl_absolute(as_ref.to_utf8()?) {
```

Comment:
`is_wsl_absolute` should probably only be called on a system with `wsl`

```suggestion
        if as_ref.is_absolute() || cfg!(target_family = "windows") && is_wsl_absolute(as_ref.to_utf8()?) {
```

Reply:
We don't currently support any Windows systems without WSL2 thought.